test/time_manager_tests.jl

```diff
@@ -0,0 +1,48 @@
+#= 
+    Unit tests for ClimaCoupler TimeManager module
+=#
+
+using Test
+using Dates
+using ClimaCoupler: Utilities, TimeManager
+
+
+@testset "test current_date" begin
+    date0 = date = DateTime("19790321", dateformat"yyyymmdd")
+    dates = (; date = [date], date0 = [date0], date1 = [Dates.firstdayofmonth(date0)])
+    tspan = (1, 90 * 86400) # Jan-Mar
+    Δt_cpl = 1 * 24 * 3600
+
+    # Fill in only the necessary parts of the simulation
+    cs = Utilities.CoupledSimulation(
+        tspan, # tspan
+        dates, # dates
+        nothing, # boundary_space
+        nothing, # fields
+        nothing, # parsed_args
+        nothing, # conservation_checks
+        0, # t
+        Δt_cpl, # Δt_cpl
+        (;), # surface_masks
+        (;), # model_sims
+        (;), # mode
+        (;), # monthly_3d_diags
+        (;), # monthly_2d_diags
+    )
+
+    for t in ((tspan[1] + Δt_cpl):Δt_cpl:tspan[end])
+        @test TimeManager.current_date(cs, t) == date0 + Dates.Second(t)
+    end
+end
+
+@testset "test strdate_to_datetime" begin
```

Comment:
What's the advantage of testing multiple cases vs just one?

Reply:
I thought it would be good to test the min and max values the string/date could have. At least for the case where the year doesn't have 4 digits, it tests that the year is correctly padded with 0s. I think we could remove the third case for the testsets without losing any information though.

Reply:
Ok, sounds good. Good thinking on testing case margins in general though!